src/SeleniumLibrary/keywords/element.py

```diff
@@ -420,14 +420,40 @@ def get_vertical_position(self, locator):
         return element.location['y']
 
     @keyword
-    def click_element(self, locator):
+    def click_element(self, locator, modifier=None):
         """Click element identified by `locator`.
 
         Key attributes for arbitrary elements are `id` and `name`. See
         `introduction` for details about locating elements.
-        """
-        self.info("Clicking element '%s'." % locator)
-        self.find_element(locator).click()
+        
+        modifier(optional) : If Element need to be opened in new window
+        
+        eg Click Element    <element_locator>    SHIFT              ----> Opens Element in new Window
+           Click Element    <element_locator>    SHIFT+CTRL         ----> Opens Element in new tab
+           CLick Element    <element_locator>                       ----> Opens Element in same window
+           
+        Note : Attempt to open element in new tab in firefox might not get window handles of all open windows, however
+        opening element in window in any browser can return window handles of all opened windows.
+        """
+
+        if(modifier!=None):
+            element = self.find_element(locator)
+            action = ActionChains(self.browser)
+            
+            modifier = modifier.strip()
+            keys = [key.strip().encode() for key in modifier.lower().split('+') ]
+
+            if ("ctrl" or "control") in keys and "shift" in keys and '' not in keys:
```

Comment:
Write unit test for keys and their combinations: SHIFT, CTRL, ALT, COMMAND. It might be easier to put the logic in separate method for unit test purposes.

Reply:
@aaltat : Please can you guide where to write unittest and any example if any ?